Sources/SwiftProtobuf/UnknownStorage.swift

```diff
@@ -44,3 +44,7 @@ public struct UnknownStorage: Equatable {
     }
   }
 }
+
+#if swift(>=5.5) && canImport(_Concurrency)
+extension UnknownStorage: @unchecked Sendable {}
```

Comment:
It may be worth us commenting here that the only reason this is `@unchecked` is that `Data` does not consistently conform to `Sendable`, and when we only support Swift versions where it does we could drop the `@unchecked`.

Reply:
Could/Should we tweak the ones generated for the Message types to only include `@unchecked` if they have a `Data` field directly?

Reply:
The limitation also applies to generated `oneof` enums that have `Data` members and extensible messages. The protocol hierarchy for extensions is otherwise `Sendable`-compatible until you get to `ProtobufBytes`. (Aside that I’m sure has been discussed elsewhere, but I hear `[UInt8]` is really nice this time of year.)

There’s also messages that use heap storage. Like `Array` In the stdlib, we can’t prove that we use COW to the compiler. Generation of `@unchecked Sendable` will likely have to remain for those.

Would you prefer a single comment that addresses all four of the limitations? “One or more of the blah types cannot blah, because blah”?

Reply:
Ooops, I see that this comment is on `UnknownStorage` and not the generated code. Trivial add to the docs 😅

Reply:
> Could/Should we tweak the ones generated for the Message types to only include `@unchecked` if they have a `Data` field directly?

Just to close the loop re: generated types, the `Data` limitation also applies to extensible messages and `oneof` enums. Then, even with that, messages with heap storage will always use `@unchecked`. So that logic would get more and more complex. I'm happy to do that — I'd rather have `Sendable` do what it's supposed to — but I want to call it out as a maintenance cost.

Reply:
Cool. Let me add a comment to the generator.